scripts/CheckSecurityConfigs.s.sol

```diff
@@ -50,130 +50,301 @@ contract CheckSecurityConfigs is Script {
     }
 
-    function runOnDir(string memory jsonDir, bool isMainnet) public {
+    function runOnDir(string memory jsonDir) public {
         VmSafe.DirEntry[] memory addressesJsonEntries = vm.readDir(jsonDir);
         hasErrors = false;
         for (uint256 i = 0; i < addressesJsonEntries.length; i++) {
-            require(bytes(addressesJsonEntries[i].errorMessage).length == 0, addressesJsonEntries[i].errorMessage);
-            runOnSingleFile(addressesJsonEntries[i].path, isMainnet);
+            require(
+                bytes(addressesJsonEntries[i].errorMessage).length == 0,
+                addressesJsonEntries[i].errorMessage
+            );
+            runOnSingleFile(addressesJsonEntries[i].path);
         }
         require(!hasErrors, "Errors occurred: See logs above for more info");
     }
 
-    function runOnSingleFile(string memory addressesJsonPath, bool isMainnet) internal {
+    function runOnSingleFile(string memory addressesJsonPath) internal {
         console2.log("Checking %s", addressesJsonPath);
+
+        bool upgradedToFPAC = chainUpgradedToFPAC(addressesJsonPath);
+
         ProtocolAddresses memory addresses = getAddresses(addressesJsonPath);
         checkAddressManager(addresses);
         checkL1CrossDomainMessengerProxy(addresses);
         checkL1ERC721BridgeProxy(addresses);
         checkL1StandardBridgeProxy(addresses);
-        checkL2OutputOracleProxy(addresses, isMainnet);
+        checkL2OutputOracleProxy(addresses, upgradedToFPAC);
         checkOptimismMintableERC20FactoryProxy(addresses);
-        checkOptimismPortalProxy(addresses);
+        checkOptimismPortalProxy(addresses, upgradedToFPAC);
         checkProxyAdmin(addresses);
         checkSystemConfigProxy(addresses);
         // TODO Check the integrity of the implementations: https://github.com/ethereum-optimism/superchain-registry/issues/33
     }
 
+    function chainUpgradedToFPAC(
+        string memory addressesJsonPath
+    ) internal pure returns (bool) {
+        return
+            keccak256(
+                abi.encodePacked(
+                    sliceString(
+                        addressesJsonPath,
+                        bytes(addressesJsonPath).length - 15,
+                        bytes(addressesJsonPath).length
+                    )
+                )
+            ) == keccak256(abi.encodePacked("sepolia/op.json"));
+    }
+
+    // Function to slice a string and return the result
+    function sliceString(
+        string memory str,
+        uint256 begin,
+        uint256 end
+    ) public pure returns (string memory) {
+        require(begin < end, "Begin index must be less than end index");
+        bytes memory strBytes = bytes(str);
+        require(end <= strBytes.length, "End index out of bounds");
+
+        bytes memory result = new bytes(end - begin);
+        for (uint i = begin; i < end; i++) {
+            result[i - begin] = strBytes[i];
+        }
+
+        return string(result);
+    }
```

Comment:
This is a fairly insane way of doing this.

I think we would save some effort by transcribing this entire test into Go. It will be much easier to maintain IMO.

Reply:
Agreed yea, we might be able to import a Solidity string library here in the future if we didn't want to transcribe.